### PR TITLE
feat: ci checks

### PR DESCRIPTION
**Pull request by raadhshenshahhaseeb**
# Description

Adds support for ci checks

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# Issues:
closes cv-68

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - '**'
+env:
+  NODEJS_VERSION: '16'
```
Comment (dogukanakkaya) [inline .github/workflows/ci.yml:11]: Shouldn't be 18?

File: .eslintrc.js [hunk 2]
```diff
@@ -6,11 +6,13 @@ module.exports = {
   },
   extends: [
     'plugin:@typescript-eslint/recommended', // Uses the recommended rules from the @typescript-eslint/eslint-plugin
```
Comment (dogukanakkaya) [inline .eslintrc.js:8]: What is the difference between this and line 11? Are they the same?

Comment (raadhshenshahhaseeb) [reply .eslintrc.js:8]: kinda but we won't need it so removed

**Review by dogukanakkaya — APPROVED** [review]
LGTM!

**Pull request merged**